src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -600,96 +869,229 @@ static class D
+    public void ParameterNameDifference()
     {
         var source = """
             using System.Runtime.CompilerServices;
             using System;
 
-            interface I1 { }
-            class C : I1 { }
-
-            static class Program
+            class C
             {
                 [Interceptable]
-                public static I1 InterceptableMethod(this I1 i1, string param) { Console.Write("interceptable " + param); return i1; }
+                public void InterceptableMethod(string s1) => throw null!;
+            }
 
+            static class Program
+            {
                 public static void Main()
                 {
                     var c = new C();
-                    c.InterceptableMethod("call site")
-                        .InterceptableMethod("call site");
+                    c.InterceptableMethod(s1: "1");
                 }
             }
 
             static class D
             {
                 [InterceptsLocation("Program.cs", 15, 11)]
-                public static I1 Interceptor1(this I1 i1, string param) { Console.Write("interceptor " + param); return i1; }
+                public static void Interceptor1(this C c, string s2) { Console.Write(s2); }
             }
             """;
-        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, expectedOutput: "interceptor call siteinterceptable call site");
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, expectedOutput: "1");
         verifier.VerifyDiagnostics();
     }
 
     [Fact]
-    public void InterceptableFromMetadata()
+    public void ParameterNamesInDifferentOrder()
     {
-        var source1 = """
+        var source = """
             using System.Runtime.CompilerServices;
             using System;
 
-            public class C
+            class C
             {
                 [Interceptable]
-                public C InterceptableMethod(string param) { Console.Write("interceptable " + param); return this; }
+                public void InterceptableMethod(string s1, string s2) => throw null!;
+            }
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    var c = new C();
+                    c.InterceptableMethod("1", "2");
+                    c.InterceptableMethod(s2: "4", s1: "3");
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 15, 11)]
+                [InterceptsLocation("Program.cs", 16, 11)]
+                public static void Interceptor1(this C c, string s2, string s1) { Console.Write(s2); Console.Write(s1); }
             }
             """;
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, expectedOutput: "1234");
+        verifier.VerifyDiagnostics();
+    }
```

Comment:
nit: consider also testing modopt or modreq differences (or adding to test plan for later). I think the way of setting this up is with modifiers in IL then overriding the method (look for `CopyMethodCustomModifiers`).

Reply:
Added note to test plan